src/main/java/hudson/plugins/analysis/core/PluginDescriptor.java

```diff
@@ -212,7 +203,12 @@ public FormValidation doCheckDefaultEncoding(@QueryParameter final String defaul
      */
     public FormValidation doCheckPattern(@AncestorInPath final AbstractProject<?, ?> project,
```

Comment:
I'm afraid this call may be invoked for the snippet generator and then cause a fatal class conversion issue